src/engine/strat_engine/engine.rs

```diff
@@ -494,9 +494,20 @@ impl Engine for StratEngine {
         name: &str,
         blockdev_paths: &[&Path],
         encryption_info: Option<&EncryptionInfo>,
+        journal_size: Option<Bytes>,
+        tag_size: Option<Bytes>,
     ) -> StratisResult<CreateAction<PoolUuid>> {
         validate_name(name)?;
         let name = Name::new(name.to_owned());
+        let rounded_journal_size = match journal_size {
```

Comment:
I think we will have to validate the tag size as well, matching our expectations to the dm-integrity expectations. I don't think there is any thing but trouble at this time that could result from allowing odd-valued tag sizes.

Reply:
I don't think this is as simple as checking for odd numbers. I'm not opposed to doing some validating, but just checking odd numbers feels incomplete. All of the values I've seen for tag size are powers of two. I think the thing to remember with tag size, however, is that that is only used to calculate the tag area which will be rounded to 4k regardless of whether the number is odd or not so I think we should either leave it as is or only support valid values of tags that integrity supports.

Reply:
I think we should only support the values that dm-integrity itself supports. We can relax that restriction at some time later if there ever appears to be any value to doing so, which seems very unlikely. It is the case that our design is to do external metadata, so integrity support in stratisd will only use this value to calculate the size of the external metadata device. But, at some time, we hope that we will be able to support integrity on NPo2 devices. If there is code that is shared for the implementation of these, then tag sizes that integrity does not support will be more of a problem. My other concern is that, at present, the D-Bus spec restricts the tag size to a byte. That means max is 128. Is that enough, or should the size be doubled?

Reply:
What I meant is: if we combine the powers of 2 restriction and the 8 digits restriction, the max possible value is 128. I also have not seen a larger tag size than 128 bytes. Is that fact enough to go on?

Reply:
ok!